internal/magic/binary.go

```diff
@@ -142,3 +142,131 @@ func Marc(raw []byte, limit uint32) bool {
 	// Field terminator is present.
 	return bytes.Contains(raw, []byte{0x1E})
 }
+
+// CborSeq matches CBOR sequences
+func CborSeq(raw []byte, limit uint32) bool {
+	if len(raw) == 0 {
+		return false
+	}
+	offset, i := 0, 0
+	ok, oldok := true, true
+	for ; ok && offset != len(raw); i++ {
+		oldok = ok
+		offset, ok = cborHelper(raw, offset)
+	}
+	if limit == uint32(len(raw)) {
+		ok = oldok
+	}
+	return ok && i > 1
+}
+
+func cborHelper(raw []byte, offset int) (int, bool) {
+	raw_len := len(raw) - offset
+	if raw_len == 0 {
+		return 0, false
+	}
+
+	mt := uint8(raw[offset] & 0xe0)
```

Comment:
I used `0x40, 0x60, 0x80...` because those are the raw byte values that CBOR uses. A bitwise AND with `0xe0` (`11100000`) has the same effect as making 5 right shifts (`>> 5`) - it discards the 5 last bits of the byte. Testing was a easier like this, but I will change it to `2, 3, 4...` to avoid confusion.

Reply:
No, `>> 5` is not the same as `& 0xe0`. https://play.golang.org/p/NFgj0z0Mt9Q